### PR TITLE
Opens a details page when clicking on a match event

### LINKED ISSUE #93: Dashboard Agenda
is it plan to clic on match in this agenda to open match board page with : 

- Match date and place
- Gymnasium address and link to navigation tools (maps, waze, ...)
- match sheet (pdf file)
- add to calendar option (ics generation?)

### PR DESCRIPTION
**Pull request by thomasFoulon**
Close #93 

Informations displayed are :
- Teams name,
- The date and the day,
- Where is the gymnasium on a mini map,
- When clicking on the map, it opens the map application and shows the location,
- A button to see the route (it also launches the map application),
- A button to add the event to the agenda on the phone.

### CODE REVIEW

**Comment by eric-taix** [comment]
Good job: the leaflet integration will also be useful for some other pages.

Question: 
- how does the user close the dialog?
- why didn't you use the common use icon for "itinéraire"? (Like the one in Google map)
- you should reduce the label for the"Ajouter a l'agenda": "Ajouter" should be enough no?
- align both buttons on the same line
- "Itinéraire" should be different than the other button: this is the défaut action

Not in this PR but we will add other informations like previous matches result, teams classifications, ... To be able to compare both teams.
So this kind of dialog is not appropriate. It's better to use a full page with a cross button on the upper left side to maximize the available display size.

Note: did you find a way to use different map styles/colors with leaflet ?

**Comment by thomasFoulon** [comment]
> how does the user close the dialog?

By clicking outside of the dialog.

> why didn't you use the common use icon for "itinéraire"? (Like the one in Google map)

I didn't find it...

> you should reduce the label for the"Ajouter a l'agenda": "Ajouter" should be enough no?
> align both buttons on the same line

I will align buttons on the same line, but I don't think "Ajouter" is enough.

> "Itinéraire" should be different than the other button: this is the défaut action

I like it that way but I can enhance the size of the "Itinéraire" button if you want.

> It's better to use a full page with a cross button on the upper left side to maximize the available display size.

I did a dialog because I think that opening a new page is too "heavy". Here the user just clicks on an event on the dashboard, opening a new page could be surprising for him and maybe too much for this action. What do you think ?

> did you find a way to use different map styles/colors with leaflet ?

I pushed this branch quite quickly, I didn't try for the moment. I'll see that soon !

**Comment by eric-taix-teads** [comment]
> By clicking outside of the dialog.

It works, but not easy for a neophyte

> I didn't find it...

`Icons.directions`

> I like it that way but I can enhance the size of the "Itinéraire" button if you want.

Don't change the size: all buttons must have the same size. In material design ui, if an action is the default action  then its background color must be different (higher emphasis). See https://material.io/components/buttons#hierarchy-and-placement 

> I did a dialog because I think that opening a new page is too "heavy". Here the user just clicks on an event on the dashboard, opening a new page could be surprising for him and maybe too much for this action. What do you think ?

I don't really understand what you mean with "too heavy": there are exactly the same number of click to open and to close a dialog, a full screen dialog and a page. Dialog should provide critical informations or ask for an action which is simple but not to display (complex) informations. So I think that a page is appropriate especially because in a near future we'll add other informations like statistics. We can also use a full screen dialog but it's usually reserved to open a final (no other interaction except a save or a dismiss action) and complex form. You should read https://material.io/components/dialogs

> I pushed this branch quite quickly, I didn't try for the moment. I'll see that soon !

👍  You can check after this PR will be merged


Note: I mentioned some links about Material Design UI. Of course we don't have to strictly follow each rule defined in this design, but Google engineers have worked a lot on the user experience and they define some guidelines for good reasons. If we don't follow a rule it must be voluntary.

**Comment by thomasFoulon** [comment]
Here is the result, it works exactly the same way than when it was a dialog (that's why there is not GIF file). What do you think ?

File: lib/pages/dashboard/widgets/timeline/gymnasium_location.dart [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+            return SafeArea(
+              child: SingleChildScrollView(
+                child: Container(
+                  child: Wrap(
+                    children: <Widget>[
+                      for (var map in availableMaps)
+                        ListTile(
+                          onTap: () {
+                            if (route) {
+                              map.showDirections(
```
Comment (eric-taix-teads) [inline lib/pages/dashboard/widgets/timeline/gymnasium_location.dart:72]: I don't know why but when I tried this branch on my phone (with 2 maps installed: Waze + Google Maps), even if I click on Waze in the bottom sheet then it always launches Google Maps...

Comment (thomasFoulon) [reply lib/pages/dashboard/widgets/timeline/gymnasium_location.dart:72]: Did you pulled the last commit ? It was a bug on the first commit ! I tried the new version with Google Maps and Waze and it well launches Waze ! The fix is exactly at the line you are showing !

**Comment by thomasFoulon** [comment]
Here is the result !

**Pull request merged**